src/tests/linalg/test_linalg.f90

```diff
@@ -56,6 +56,24 @@ program test_linalg
   call test_trace_int32
   call test_trace_int64
 
+  !
+  ! outer product
+  !
```

Comment:
You haven't implemented the tests yet right? In either case, the overall structure is fine. I would suggest some simple cases with vectors of length 2, 3, and 4:

```
u = [1.,2.]
v = [1.,3.]
expected = reshape([1.,3.,2.,6.],[2,2])
abs_diff = abs(expected - outer_product(u,v)) 
call check(all(abs_diff < sptol), ...)
```